src/Jh/DataImportMagento/Writer/ShipmentWriter.php

```diff
@@ -64,6 +70,17 @@ public function writeItem(array $item)
                 ->addObject($shipment->getOrder())
                 ->save();
 
+            if (isset($item['tracking_number'], $item['carrier'])) {
+                $tracking = clone $this->trackingModel;
+                $tracking
+                    ->setShipment($shipment)
+                    ->setData('title', $item['carrier'])
+                    ->setData('number', $item['tracking_number'])
+                    ->setData('carrier_code', 'custom')
+                    ->setData('order_id', $shipment->getData('order_id'))
+                    ->save();
+                \Mage::dispatchEvent('after_shipping_data_save', ['item' => $item]);
```

Comment:
Okay here is the current client side code the sends an email based on the event dispatch

`public function dispatchShippingEmail(Varien_Event_Observer $e)
    {
        $item           = $e->getData('item');
        $order          = Mage::getModel('sales/order')->loadByIncrementId($item['order_id']);
        $customerEmail  = $order->getCustomerEmail();
        $shipments      = $order->getShipmentsCollection();

        foreach ($shipments as $currentShipment) {
            if ($customerEmail) {
                $currentShipment->sendEmail(
                    true,
                    sprintf(
                        'Your Order has been shipped by %s and your tracking number is %s',
                        $item['carrier'],
                        $item['tracking_number']
                    )
                );
                $currentShipment->setEmailSent(true);
            }
        }
    }`

As you can see I'll need access to `order_id, `tracking_number` and `carrier` from `$item`

I'll also need access to $emailFlag from ShipmentWriter.php (its not been created yet) which will be a bool for checking if email sending is set. 

What would be the best place to put this method needing access to this information. A method in ShippingWriter or would I be able to access this from ShippingWriterFactory.php as a constructor argument.